pkg/config/ledger_config.go

```diff
@@ -14,6 +14,9 @@ type Ledger struct {
 	KeepOnlyLatestState bool `yaml:"KeepOnlyLatestState"`
 	// RemoveUntraceableBlocks specifies if old data should be removed.
 	RemoveUntraceableBlocks bool `yaml:"RemoveUntraceableBlocks"`
+	// RemoveUntraceableHeaders is used in addition to RemoveUntraceableBlocks
+	// when headers need to be removed as well.
+	RemoveUntraceableHeaders bool `yaml:"RemoveUntraceableHeaders"`
```

Comment:
I think we need to include this extension to `result.Protocol` RPC structure, because some applications may be affected by this change. RPC server/client user need to distinguish "light" nodes from "archive" nodes somehow.

Reply:
Node with `RemoveUntraceableBlocks` is already not an archive one, so `RemoveUntraceableHeaders` don't change a lot here. This makes sense as a separate issue.

Reply:
https://github.com/nspcc-dev/neo-go/issues/3754